### PR TITLE
Convert Intrinsic Ratio from nsSize to Float

### PR DESCRIPTION
**Pull request by RealityRipple**
First of two-phase implementation of Intrinsic Aspect Ratios from `width=` and `height=` HTML values:

Any use of `nsSize` for Aspect Ratio values now uses a `float` value. Some special helper functions simplify basic aspect ratio tasks, both in current use and in phase two of Intrinsic Aspect Ratio support.

This code should cause no real functional changes as of yet. Resolves issue #1619.

### CODE REVIEW

**Comment by mattatobin** [comment]
You need to fix the commit message for the second in this PR

**Comment by RealityRipple** [comment]
I did not mean for the third one to get pushed yet, either... I'll clean it up as soon as the compile I'm working on is done.

**Comment by wolfbeast** [comment]
The dimension computation was likely added by Mozilla in a different bug. It can probably be found by checking Mozilla's `gecko-dev` blame, but since this is a logical thing to do (set the missing value if the 2 dependent values exist) I don't see the point in spending the extra time.

File: layout/generic/nsFlexContainerFrame.cpp [hunk 1]
```diff
@@ -1520,22 +1520,20 @@ CrossSizeToUseWithRatio(const FlexItem& aFlexItem,
...
-                        const nsSize& aIntrinsicRatio,
+                        const AspectRatio& aIntrinsicRatio,
                         const FlexboxAxisTracker& aAxisTracker)
 {
-  MOZ_ASSERT(aAxisTracker.GetCrossComponent(aIntrinsicRatio) != 0,
+  MOZ_ASSERT(aIntrinsicRatio,
              "Invalid ratio; will divide by 0! Caller should've checked...");
+  AspectRatio ratio = aAxisTracker.IsMainAxisHorizontal()
+                          ? aIntrinsicRatio
+                          : aIntrinsicRatio.Inverted();
```
Comment (wolfbeast) [inline layout/generic/nsFlexContainerFrame.cpp:1534]: Nit: code style
```
somelongvariable = somelongexpression ?
                   somevalue1 :
                   somevalue2;
```

File: layout/generic/nsHTMLCanvasFrame.cpp [hunk 2]
```diff
@@ -211,10 +196,15 @@ nsHTMLCanvasFrame::GetIntrinsicSize()
...
-/* virtual */ nsSize
+/* virtual */ AspectRatio
 nsHTMLCanvasFrame::GetIntrinsicRatio()
 {
-  return IntrinsicRatioFromCanvasSize(GetCanvasSize());
+/*
+  if (StyleDisplay()->IsContainSize()) {
+    return AspectRatio();
+  }
+ */
```
Comment (wolfbeast) [inline layout/generic/nsHTMLCanvasFrame.cpp:206]: Please add a comment that this needs to be fixed when containsize is implemented.

File: layout/generic/nsVideoFrame.cpp [hunk 3]
```diff
@@ -533,7 +531,9 @@ nsVideoFrame::ComputeSize(nsRenderingContext *aRenderingContext,
   intrinsicSize.height.SetCoordValue(size.height);
 
   // Only video elements have an intrinsic ratio.
-  nsSize intrinsicRatio = HasVideoElement() ? size : nsSize(0, 0);
+  auto intrinsicRatio = HasVideoElement()
+                            ? AspectRatio::FromSize(size)
+                            : AspectRatio();
```
Comment (wolfbeast) [inline layout/generic/nsVideoFrame.cpp:536]: Nit: code style

File: layout/xul/nsImageBoxFrame.cpp [hunk 4]
```diff
@@ -402,12 +402,13 @@ nsImageBoxFrame::PaintImage(nsRenderingContext& aRenderingContext,
...
     IntrinsicSize intrinsicSize;
-    nsSize intrinsicRatio;
+    AspectRatio intrinsicRatio;
     if (mIntrinsicSize.width > 0 && mIntrinsicSize.height > 0) {
       // Image has a valid size; use it as intrinsic size & ratio.
       intrinsicSize.width.SetCoordValue(mIntrinsicSize.width);
       intrinsicSize.height.SetCoordValue(mIntrinsicSize.height);
-      intrinsicRatio = mIntrinsicSize;
+      intrinsicRatio =
+          AspectRatio::FromSize(mIntrinsicSize);
```
Comment (wolfbeast) [inline layout/xul/nsImageBoxFrame.cpp:411]: Nit: please make one line

File: layout/generic/nsHTMLCanvasFrame.cpp (hunk 2 above)
Comment (RealityRipple) [reply layout/generic/nsHTMLCanvasFrame.cpp:207]: I figured 'contain: size' would be enough to search for in the future, but if there's a better label or "TODO:" prefix you'd like me to add, let me know. There's gonna be a lot more of these in the next PR.

Comment (wolfbeast) [reply layout/generic/nsHTMLCanvasFrame.cpp:207]: That comment is fine to copypasta on all commented-out blocks as far as I am concerned.

**Pull request merged**